### PR TITLE
Added paper links in logitprocess.py

### PR DESCRIPTION
**Pull request by pranith7**
# What does this PR do?

Fixes # (issue)
#24783 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

### CODE REVIEW

**Comment by sgugger** [comment]
cc @gante 

Thanks for your PR! Please run `make style` on your branch to fix the auto-formatting issues.

File: src/transformers/generation/logits_process.py [hunk 1]
```diff
@@ -1333,6 +1333,8 @@ class WhisperTimeStampLogitsProcessor(LogitsProcessor):
     Whisper specific Processor. This processor can be used to force a list of tokens. The processor will set their log
     probs to `inf` so that they are sampled at their corresponding index.
 
+    See [the paper](https://arxiv.org/abs/2303.00747) for more information.
```
Comment (gante) [inline src/transformers/generation/logits_process.py:1336]: The paper here would be the original whisper paper, https://arxiv.org/abs/2212.04356 :)

The timestamps returned here are the sequence-level timestamps, not the word-level timestamps of WhisperX. (As further proof, this code was added before the WhisperX paper came out, as a result of a refactor)

File: src/transformers/generation/logits_process.py [hunk 2]
```diff
@@ -1399,6 +1401,8 @@ class ClassifierFreeGuidanceLogitsProcessor(LogitsProcessor):
     correspond to the unconditional logits (predicted from an empty or 'null' prompt). The processor computes a
     weighted average across the conditional and unconditional logits, parameterised by the `guidance_scale`.
 
+    See [the paper](https://arxiv.org/abs/2306.17806) for more information.
```
Comment (gante) [inline src/transformers/generation/logits_process.py:1404]: This is also not the paper used here :)

The correct paper is https://arxiv.org/abs/2306.05284 (similar techniques, but not exactly the same, the one you linked can be seen as an upgrade to this technique)

**Comment by pranith7** [comment]
cc @gante  updated the links for the paper

**Review by gante — APPROVED** [review]
Thank you for iterating 💪

**Pull request merged**